lib/Command/TransferLegacyRows.php

```diff
@@ -0,0 +1,178 @@
+use OCA\Tables\Db\Table;
+use OCA\Tables\Errors\InternalError;
+use OCA\Tables\Errors\NotFoundError;
+use OCA\Tables\Errors\PermissionError;
+use OCA\Tables\Service\ColumnService;
+use OCA\Tables\Service\TableService;
+use OCP\DB\Exception;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class TransferLegacyRows extends Command {
+	protected TableService $tableService;
+	protected LoggerInterface $logger;
+	protected LegacyRowMapper $legacyRowMapper;
+	protected RowMapper $rowMapper;
+	protected ColumnService $columnService;
+
+	public function __construct(TableService $tableService, LoggerInterface $logger, LegacyRowMapper $legacyRowMapper, RowMapper $rowMapper, ColumnService $columnService) {
+		parent::__construct();
+		$this->tableService = $tableService;
+		$this->logger = $logger;
+		$this->legacyRowMapper = $legacyRowMapper;
+		$this->rowMapper = $rowMapper;
+		$this->columnService = $columnService;
+	}
+
+	protected function configure(): void {
+		$this
+			->setName('tables:legacy:transfer:rows')
+			->setDescription('Transfer table legacy rows to new schema.')
+			->addArgument(
+				'table-ids',
+				InputArgument::OPTIONAL,
+				'IDs of tables for the which data is to be transferred. (Multiple comma seperated possible)'
+			)
+			->addOption(
+				'all',
+				null,
+				InputOption::VALUE_NONE,
+				'Transfer all table data.'
+			)
+			->addOption(
+				'no-delete',
+				null,
+				InputOption::VALUE_OPTIONAL,
+				'Set to not delete data from new db structure if any.'
+			)
+		;
+	}
+
+	/**
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @return int
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		$tableIds = $input->getArgument('table-ids');
+		$optionAll = !!$input->getOption('all');
+		$optionNoDelete = $input->getOption('no-delete') ?: null;
+
+		if ($optionAll) {
+			$output->writeln("Look for tables");
+			try {
+				$tables = $this->tableService->findAll('', true, true, false);
+				$output->writeln("Found ". count($tables) . " table(s)");
+			} catch (InternalError $e) {
+				$output->writeln("Error while fetching tables. Will aboard.");
+				return 1;
+			}
+		} elseif ($tableIds) {
+			$output->writeln("Look for given table(s)");
+			$tableIds = explode(',', $tableIds);
+			$tables = [];
+			foreach ($tableIds as $tableId) {
+				try {
+					$tables[] = $this->tableService->find((int)ltrim($tableId), true, '');
+				} catch (InternalError|NotFoundError|PermissionError $e) {
+					$output->writeln("Could not load table id " . $tableId . ". Will continue.");
+				}
+			}
+		} else {
+			$output->writeln("🤷🏻‍ Add at least one table id or add the option --all to transfer all tables.");
+			return 1;
```

Comment:
```suggestion
			return 2;
```

Nitpick. In case anyone would want to interpret exit codes :)